lib/connection/connection.js

```diff
@@ -136,7 +136,8 @@ function Connection(context)
   {
     return new Promise((resolve, reject) =>
     {
-      this.heartbeat((err, body) => err ? reject(err) : resolve(body));
+      // previous version of driver called `select 1;` which result in `[ { '1': 1 } ]`
+      this.heartbeat((err) => err ? reject(err) : resolve([ { '1': 1 } ]));
```

Comment:
why the new code care about previous version of driver?

Reply:
method `heartbeatAsync` was available in previous version of driver (the new method we added is `isValidAsync`) and someone could use this method and parse output from succesfully resolved promise - which is different now since it is using SF heartbeat, not `Select 1`